cli/pcluster/config/param_types.py

```diff
@@ -0,0 +1,1202 @@
+    merging info from "initial_queue_size" and "maintain_initial_size" when the scheduler is a traditional one.
+    """
+
+    def _init_from_cfn(self, cfn_params=None, cfn_value=None):
+        """Initialize param value by parsing the right CFN input according to the scheduler."""
+        cfn_converter = self.map.get("cfn", None)
+        if cfn_converter and cfn_params:
+            cfn_value = get_cfn_param(cfn_params, cfn_converter) if cfn_converter else "NONE"
+            # initialize the value from cfn or from map according to the scheduler
+            if get_cfn_param(cfn_params, "Scheduler") == "awsbatch":
+                self.value = self.get_value_from_string(cfn_value)
+            else:
+                self._init_from_map()
+        elif cfn_value:
+            self.value = self.get_value_from_string(cfn_value)
+        else:
+            self._init_from_map()
+
+    def to_cfn(self):
+        """Convert parameter to CFN representation."""
+        cfn_params = {}
+
+        cluster_config = self.pcluster_config.get_section(self.section_key)
+        if cluster_config.get_param_value("scheduler") == "awsbatch":
+            cfn_value = cluster_config.get_param_value("min_vcpus")
+            cfn_params.update({self.map.get("cfn"): str(cfn_value)})
+
+        return cfn_params
+
+
+class AvailabilityZoneParam(Param):
+    """
+    Class to manage master_availability_zone internal attribute.
+
+    This parameter is not exposed as configuration parameter in the file but it exists as CFN parameter
+    and it is used during EFS conversion and validation.
+    """
+
+    def _init_from_file(self, config_parser):
+        """Initialize the Availability zone of the cluster by checking the Master Subnet."""
+        section_name = _get_file_section_name(self.section_key, self.section_label)
+        master_subnet_id = config_parser.get(section_name, "master_subnet_id")
+        self.value = get_avail_zone(master_subnet_id)
+        self._check_allowed_values()
+
+    def to_file(self, config_parser):
+        """Do nothing, because master_availability_zone it is an internal parameter, not exposed in the config file."""
+        pass
+
+
+# ---------------------- SettingsParam ---------------------- #
+
+
+class SettingsParam(Param):
+    """Class to manage *_settings parameter on which the value is a single value (e.g. vpc_settings = default)."""
+
+    def __init__(
+        self,
+        section_key,
+        section_label,
+        param_key,
+        param_map,
+        pcluster_config,
+        cfn_value=None,
+        config_parser=None,
+        cfn_params=None,
+    ):
+        """Extend Param by adding info regarding the section referred by the settings."""
+        self.related_section_map = param_map.get("referred_section")
+        self.related_section_key = self.related_section_map.get("key")
+        self.related_section_type = self.related_section_map.get("type")
+        super(SettingsParam, self).__init__(
```

Comment:
minor: it could be a good practice to call super as first thing in the constructor

Reply:
fixed.